File: template/base_feature/src/main/java/ru/surfstudio/standard/ui/widget/di/WidgetScreenConfigurator.java
package ru.surfstudio.standard.ui.widget.di;

import ru.surfstudio.android.core.ui.activity.CoreActivityInterface;
import ru.surfstudio.android.mvp.widget.configurator.BaseWidgetViewConfigurator;
import ru.surfstudio.android.mvp.widget.delegate.ViewContextUnwrapper;
import ru.surfstudio.standard.ui.activity.di.ActivityComponent;
import ru.surfstudio.standard.ui.widget.di.WidgetScreenModule;

/**
 * Базовый конфигуратор для WidgetView
 */

public abstract class WidgetScreenConfigurator
        extends BaseWidgetViewConfigurator<ActivityComponent, WidgetScreenModule> {

    @Override
    protected WidgetScreenModule getWidgetScreenModule() {
        return new WidgetScreenModule(getPersistentScope());
    }

    @Override
    protected ActivityComponent getParentComponent() {
        return (ActivityComponent) (
                ViewContextUnwrapper.unwrapContext(
                        getTargetWidgetView().getContext(),
                        CoreActivityInterface.class
                )
        )
                .getPersistentScope()
                .getConfigurator()
                .getActivityComponent();
    }
}

File: template/base_feature/src/main/java/ru/surfstudio/standard/ui/widget/di/WidgetScreenConfigurator.kt
package ru.surfstudio.standard.ui.widget.di

import ru.surfstudio.android.core.ui.activity.CoreActivityInterface
import ru.surfstudio.android.mvp.widget.configurator.BaseWidgetViewConfigurator
import ru.surfstudio.android.mvp.widget.delegate.unwrapContext
import ru.surfstudio.standard.ui.activity.di.ActivityComponent

/**
 * Базовый конфигуратор для WidgetView
 */

abstract class WidgetScreenConfigurator : BaseWidgetViewConfigurator<ActivityComponent, WidgetScreenModule>() {

    override fun getWidgetScreenModule(): WidgetScreenModule {
        return WidgetScreenModule(persistentScope)
    }

    override fun getParentComponent(): ActivityComponent {
        return unwrapContext<CoreActivityInterface>(targetWidgetView.context)
                .getPersistentScope()
                .getConfigurator()
                .getActivityComponent() as ActivityComponent
    }
}